lib/chef/win32/registry.rb

```diff
@@ -91,8 +91,15 @@ def delete_value(key_path, value)
             return true
           end
           hive.open(key, ::Win32::Registry::KEY_SET_VALUE | registry_system_architecture) do |reg|
-            reg.delete_value(value[:name])
-            Chef::Log.debug("Deleted value #{value[:name]} from registry key #{key_path}")
+            # ::Win32::Registry#delete_value is broken in Ruby 2.1 (up to Ruby 2.1.6p336).
+            # This should be resolved a later release (see note #9 in link below).
+            # https://bugs.ruby-lang.org/issues/10820
+            if RegDeleteValueW(reg.hkey, wstring(value[:name]))
+              Chef::Log.debug("Deleted value #{value[:name]} from registry key #{key_path}")
+            else
+              Chef::Log.debug("Unable to delete value #{value[:name]} from registry key #{key_path}")
+              return false
+            end
```

Comment:
@chefsalim (re: https://github.com/chef/chef/pull/3849#issuecomment-136494592) isn't that what's happening here? 0 is truthy, non-zero is falsey.